### PR TITLE
feat: add unit tests for LeftNav component

### PR DESCRIPTION
**Pull request by Dripcoding**
Add unit tests for the LeftNav component

### CODE REVIEW

File: facade-app/src/test/LeftNav.test.tsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+    it.each(inputs)(`should have sub item %s`, (text) => {
+      const mock = () => jest.fn();
+      render(<LeftNav globalState={testFixture} setGlobalState={mock} />);
+
+      const item = screen.getByText(text);
+      expect(item).toBeInTheDocument();
+    });
+
+    const levels = [
```
Comment (preetkaran20) [inline facade-app/src/test/LeftNav.test.tsx:69]: This data will keep on changing so let's not add check on the vulnerabilities.

File: facade-app/src/test/LeftNav.test.tsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+      it.each(inputs)(`should have sub item %s`, (text) => {
+          const mock = () => jest.fn();
+          render(<LeftNav globalState={testFixture} setGlobalState={mock} />);
+
+          const item = screen.getAllByText(text)[0]
+          expect(item).toBeInTheDocument()
+      })
+
+      const levels = [
+          ['VulnerableApp-jsp.FileUpload.LEVEL_1', 'LEVEL_1'],
```
Comment (preetkaran20) [inline facade-app/src/test/LeftNav.test.tsx:188]: Same for this one as well. This is only available if VulnerableApp-jsp is integrated so let's not have check on these values.

File: facade-app/src/test/LeftNav.test.tsx [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        it.each(inputs)(`should have sub item %s`, (text) => {
+            const mock = () => jest.fn();
+            render(<LeftNav globalState={testFixture} setGlobalState={mock} />)
+
+            const item = screen.getAllByText(text)[0]
+            expect(item).toBeInTheDocument()
+        })
+
+        const levels = [
+            ['VulnerableApp-php.FileUpload.LEVEL_1', 'LEVEL_1'],
```
Comment (preetkaran20) [inline facade-app/src/test/LeftNav.test.tsx:231]: Same here

File: facade-app/src/test/LeftNav.test.tsx (hunk 1 above)
Comment (preetkaran20) [reply facade-app/src/test/LeftNav.test.tsx:69]: or is it mocked data from fixtures.ts?

File: facade-app/src/test/LeftNav.test.tsx (hunk 2 above)
Comment (preetkaran20) [reply facade-app/src/test/LeftNav.test.tsx:188]: or is it mocked data from fixtures.ts?

File: facade-app/src/test/LeftNav.test.tsx (hunk 3 above)
Comment (preetkaran20) [reply facade-app/src/test/LeftNav.test.tsx:231]: or is it mocked data from fixtures.ts?

File: facade-app/src/test/LeftNav.test.tsx (hunk 1 above)
Comment (Dripcoding) [reply facade-app/src/test/LeftNav.test.tsx:69]: yes these are mocked from the fixtures.ts file

File: facade-app/src/test/LeftNav.test.tsx (hunk 2 above)
Comment (Dripcoding) [reply facade-app/src/test/LeftNav.test.tsx:188]: these are mocked from fixtures.ts too

File: facade-app/src/test/LeftNav.test.tsx (hunk 3 above)
Comment (Dripcoding) [reply facade-app/src/test/LeftNav.test.tsx:231]: these are also mocked from fixtures.ts

**Review by preetkaran20 — APPROVED** [review]

**Comment by preetkaran20** [comment]
Thank you!!! merging the PR.

**Pull request merged**